### PR TITLE
Fix issue searching for android dependencies 

### PR DESCRIPTION
**Pull request by marktefftech**
Currently we are searching in /system/lib64 for dependencies but fuzzer deps get pushed to the /data/fuzz directory. Also  clearing out downloaded deps before fuzzing

### CODE REVIEW

File: src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py [hunk 1]
```diff
@@ -872,13 +872,17 @@ def _get_default_args(self, executable_path, extra_args):
     """Return a set of default arguments to pass to adb binary."""
     default_args = ['shell']
 
-    # Add directory containing libclang_rt.ubsan_standalone-aarch64-android.so
-    # to LD_LIBRARY_PATH.
+    # LD_LIBRARY_PATH set to search for fuzzer deps first,
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py:875]: nit: change to:
```
    # LD_LIBRARY_PATH set to search for fuzzer deps first, and then sanitizers
    # if any are found
```

File: src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py [hunk 2]
```diff
@@ -872,13 +872,17 @@ def _get_default_args(self, executable_path, extra_args):
...
+    # and then sanitizers if any are found
     ld_library_path = ''
     if not android.settings.is_automotive():
       # TODO(MHA3): Remove this auto check.
-      ld_library_path = android.sanitizer.get_ld_library_path_for_sanitizers()
-    if ld_library_path:
+      exe_path_dir = os.path.dirname(executable_path)
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py:880]: nit: just call this variable `executable_dir`

File: src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py [hunk 3]
```diff
@@ -872,13 +872,17 @@ def _get_default_args(self, executable_path, extra_args):
...
+      deps_path = self._get_device_path(exe_path_dir) + '/lib'
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py:881]: Can we use os.path.join(self._get_device_path(executable_dir), 'lib') instead?

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Review by jonathanmetzman — CHANGES_REQUESTED** [review]
please fix the nits before i merge.

File: src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py (hunk 2 above)
Comment (marktefftech) [reply src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py]: Done.

File: src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py (hunk 3 above)
Comment (marktefftech) [reply src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py]: Agreed this is a better approach, done.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Pull request merged**